bitacross-worker/bitacross/core/bc-relayer-registry/src/lib.rs

```diff
@@ -0,0 +1,197 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+extern crate core;
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+extern crate sgx_tstd as std;
+
+#[cfg(all(feature = "std", feature = "sgx"))]
+compile_error!("feature \"std\" and feature \"sgx\" cannot be enabled at the same time");
+
+use sp_std::{boxed::Box, fmt::Debug};
+
+use lazy_static::lazy_static;
+use litentry_primitives::Identity;
+use log::error;
+use std::{collections::BTreeMap, path::PathBuf, sync::Arc};
+pub use teerex_primitives::{decl_rsa_request, ShardIdentifier, SidechainBlockNumber};
+
+#[cfg(feature = "std")]
+use std::sync::RwLock;
+#[cfg(feature = "sgx")]
+use std::sync::SgxRwLock as RwLock;
+
+lazy_static! {
+	/// Global instance of a RelayerRegistry
+	pub static ref GLOBAL_RELAYER_REGISTRY: Arc<RelayerRegistry> = Default::default();
+}
+
+pub type RelayerRegistryMap = BTreeMap<Identity, ()>;
+
+#[derive(Default)]
+pub struct RelayerRegistry {
+	pub registry: RwLock<RelayerRegistryMap>,
+	pub seal_path: PathBuf,
+}
+
+pub type RegistryResult<T> = core::result::Result<T, RegistryError>;
+
+#[cfg(feature = "sgx")]
+use thiserror_sgx as thiserror;
+
+#[derive(Debug, thiserror::Error)]
+pub enum RegistryError {
+	#[error("poison lock")]
+	PoisonLock,
+	#[error("empty Relayer registry")]
+	EmptyRegistry,
+	#[error(transparent)]
+	Other(#[from] Box<dyn std::error::Error + Sync + Send + 'static>),
+}
+
+impl From<std::io::Error> for RegistryError {
+	fn from(e: std::io::Error) -> Self {
+		Self::Other(e.into())
+	}
+}
+
+impl From<codec::Error> for RegistryError {
+	#[cfg(feature = "std")]
+	fn from(e: codec::Error) -> Self {
+		Self::Other(e.into())
+	}
+
+	#[cfg(feature = "sgx")]
+	fn from(e: codec::Error) -> Self {
+		Self::Other(std::format!("{:?}", e).into())
+	}
+}
+
+#[cfg(feature = "sgx")]
+mod sgx {
+	pub use codec::{Decode, Encode};
+	// TODO: Update to relayer registry file
```

Comment:
Are you going to resolve this TODO in this PR or following one ?

Reply:
Updated ✅